src/app/mesh/qgsmaptooleditmeshframe.h

```diff
@@ -0,0 +1,180 @@
+    void cadCanvasPressEvent( QgsMapMouseEvent *e ) override;
+    void cadCanvasReleaseEvent( QgsMapMouseEvent *e ) override;
+    void cadCanvasMoveEvent( QgsMapMouseEvent *e ) override;
+    void keyPressEvent( QKeyEvent *e ) override;
+    void keyReleaseEvent( QKeyEvent *e ) override;
+
+    //! Start addition of a new vertex on double-click
+    void canvasDoubleClickEvent( QgsMapMouseEvent *e ) override;
+
+  private slots:
+    void setCurrentLayer( QgsMapLayer *layer );
+    void onEdit();
+    void onEdidingStarted();
+    void onEditingStopped();
+
+  private:
+    // methods
+    const QgsMeshVertex mapVertex( int index ) const;
+    const QgsMeshFace nativeFace( int index ) const;
+
+    void highlightCurrentHoveredFace( const QgsPointXY &mapPoint );
+    void highlightCloseVertex( const QgsPointXY &mapPoint );
+
+    void createZValueWidget();
+    void deleteZvalueWidget();
+    void clear();
+    void addVertex( const QgsPointXY &mapPoint, const QgsPointLocator::Match &mapPointMatch );
+    void updateFreeVertices();
+
+    //! Checks if we are closed to a vertex, if yes return the index of the vertex;
+    int closeVertex( const QgsPointXY &point ) const;
+
+    QgsPointSequence nativeFaceGeometry( int faceIndex ) const;
+
+    QgsPointXY newFaceMarkerPosition( int vertexIndex );
+
+    void addVertexToFaceCanditate( int vertexIndex );
+    bool testNewVertexInFaceCanditate( int vertexIndex );
+
+    // selection private method
+    void setSelectedVertex( const QList<int> newSelectedVertex, bool ctrl );
+    void clearSelectedvertex();
+    void removeSelectedVerticesFromMesh();
+    void removeSelectedVerticesFromMesh( bool fillHole );
+    void selectVerticesInGeometry( const QgsGeometry &geometry, bool ctrl );
+
+    void applyZValueOnSelectedVertices();
+
+    // members
+    enum State
+    {
+      Default,
+      AddingNewFace,
+      Selecting,
+    };
+
+    State mCurrentState = Default;
+
+    QPointer<QgsMeshLayer> mCurrentLayer = nullptr;
+    QPointer<QgsMeshEditor> mCurrentEditor = nullptr;
+    std::unique_ptr<QgsSnapIndicator> mSnapIndicator;
+    int mCurrentFaceIndex = -1;
+    int mCurrentVertexIndex = -1;
+    QList<int> mNewFaceCandidate;
+    QList<int> mSelectedVertex;
+    bool mDoubleClick = false;
+    double mCurrentZValue = 0;
+
+    //! Rubber band used to highlight a face that is on mouse over and not dragging anything
+    QgsRubberBand *mFaceRubberBand = nullptr;
+    //! Rubber band used to highlight vertex of the face that is on mouse over and not dragging anything
+    QgsRubberBand *mFaceBandMarkers = nullptr;
+    //! Rubber band used to highlight the vertex that is in mouse over and not dragging anything
+    QgsRubberBand *mVertexBand = nullptr;
+    //! Marker used to propose to add a new face when a boundary vertex is higthlight
+    QgsVertexMarker *mNewFaceMarker = nullptr;
+    //! Rubber band used when adding a new face
+    QgsRubberBand *mNewFaceBand = nullptr;
```

Comment:
is it own by this map tool? smart pointers ?

Reply:
rubber band and marker are own by the QgsMapCanvas instance